### PR TITLE
Add align, nofree and nonnull to vmctx pointer. Dereferenceable is still TODO.

### PR DESCRIPTION
**Pull request by nlewycky**
This gives LLVM more information about the vmctx pointer which it can use for better code generation and faster reasoning while compiling.

### CODE REVIEW

File: lib/compiler-llvm/src/abi/aarch64_systemv.rs [hunk 1]
```diff
@@ -47,19 +49,36 @@ impl Abi for Aarch64SystemV {
...
         Ok(match sig.results() {
             [] => (
                 intrinsics
                     .void_ty
                     .fn_type(&param_types.collect::<Result<Vec<_>, _>>()?, false),
-                vec![],
+                vmctx_attributes
+                    .into_iter()
+                    .map(|x| (x, AttributeLoc::Param(0)))
+                    .collect(),
```
Comment (syrusakbary) [inline lib/compiler-llvm/src/abi/aarch64_systemv.rs:71]: This code seems repeated across the different match branches.
Why don't we move it up so we only write it once?

File: lib/compiler-llvm/src/abi/x86_64_systemv.rs [hunk 2]
```diff
@@ -67,14 +81,20 @@ impl Abi for X86_64SystemV {
                 intrinsics
                     .void_ty
                     .fn_type(&param_types.collect::<Result<Vec<_>, _>>()?, false),
-                vec![],
+                vmctx_attributes
+                    .into_iter()
+                    .map(|x| (x, AttributeLoc::Param(0)))
+                    .collect(),
```
Comment (syrusakbary) [inline lib/compiler-llvm/src/abi/x86_64_systemv.rs:87]: This code seems repeated across the different match branches.
Why don't we move it up so we only write it once?

File: lib/compiler-llvm/src/abi/aarch64_systemv.rs (hunk 1 above)
Comment (nlewycky) [reply lib/compiler-llvm/src/abi/aarch64_systemv.rs]: Done.

File: lib/compiler-llvm/src/abi/x86_64_systemv.rs (hunk 2 above)
Comment (nlewycky) [reply lib/compiler-llvm/src/abi/x86_64_systemv.rs]: Done.

**Review by syrusakbary — APPROVED** [review]

**Review by MarkMcCaskey — APPROVED** [review]

**Pull request merged**